File: rpud/src/zzz.c

/*
 Authors 
 Alexander Freudenberg, afreuden@mail.uni-mannheim.de

Courtesy of Martin Schlather

 Copyright (C) 2021, Martin Schlather, Alexander Freudenberg

This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 3
of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, writne to the Free Software
Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.  
*/

#include <R_ext/Rdynload.h>
#include "rgpu.h"


#define CALLDEF_DO(name, n) {#name, (DL_FUNC) &name, n}
static R_CallMethodDef callMethods[] = {
   CALLDEF_DO(cu_matmul,3),
   CALLDEF_DO(access_pointer,1),
    {NULL, NULL, 0}
};

void R_init_rgpu(DllInfo *dll) {
    R_registerRoutines(dll,
            NULL,
            callMethods, // defined below
            NULL, // .Fortran
            NULL); // ext
    R_useDynamicSymbols(dll, FALSE);
}

void R_unload_rgpu(DllInfo *info) { }

File: miraculix/R/zzz.R

## Authors 
## Martin Schlather, schlather@math.uni-mannheim.de
##
##
## Copyright (C) 2017 -- 2019 Martin Schlather
##
## This program is free software; you can redistribute it and/or
## modify it under the terms of the GNU General Public License
## as published by the Free Software Foundation; either version 3
## of the License, or (at your option) any later version.
##
## This program is distributed in the hope that it will be useful,
## but WITHOUT ANY WARRANTY; wit502hout even the implied warranty of
## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
## GNU General Public License for more details.
##
## You should have received a copy of the GNU General Public License
## along with this program; if not, write to the Free Software
## Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.  


.onLoad <- function(lib, pkg) {
  ## DESCRIPTION MUSS RandomFieldsUtils EINGEFUEGT SEIN
  .C("loadoptions")
  # n <- 0.75 * cores() + 0.25 * cpus()
  n<-4
  RFoptions(solve.solve_method=c("cholesky", "cholesky"),
            solve.max_chol=10^5, ## OK
            basic.cores = if (is.na(n)) 1 else as.integer(n))
}


.onAttach <- function (lib, pkg) {
}

.onDetach <- function(lib) {
## do not use the following commmands in .onDetach!
#  RFoptions(storing=FALSE) ## delete everything
#  .C("detachRFoptionsRandomFields")
}

.onUnload <- function(lib, pkg){
#   RFoptions(storing=FALSE) ## delete everything
  .C("detachoptions")
}
